include/nbl/builtin/hlsl/sampling/basic.hlsl

```diff
@@ -0,0 +1,44 @@
+// Copyright (C) 2018-2025 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_BUILTIN_HLSL_SAMPLING_BASIC_INCLUDED_
+#define _NBL_BUILTIN_HLSL_SAMPLING_BASIC_INCLUDED_
+
+#include <nbl/builtin/hlsl/cpp_compat.hlsl>
+#include <nbl/builtin/hlsl/ieee754.hlsl>
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace sampling
+{
+
+template<typename T NBL_PRIMARY_REQUIRES(concepts::FloatingPointLikeScalar<T>)
+struct PartitionRandVariable
+{
+    using floating_point_type = T;
+    using uint_type = typename unsigned_integer_of_size<sizeof(floating_point_type)>::type;
```

Comment:
continues https://github.com/Devsh-Graphics-Programming/Nabla/pull/959#discussion_r2588406305

use `unsigned_integer_of_size_t<N>` instead of `typename unsigned_integer_of_size<N>::type`